### PR TITLE
Trigger a reconnect immediately on ping timeout

### PR DESCRIPTION
**Pull request by connor4312**

### CODE REVIEW

File: lib/ws/ws.js [hunk 1]
```diff
@@ -282,35 +294,35 @@ BeamSocket.prototype.boot = function () {
...
+        this.unspool.apply(this, arguments);
     });
 
     // We got an incoming data packet.
-    ws.on('message', function () {
-        self._resetPingTimeout();
-        self.parsePacket.apply(self, arguments);
-    });
+    ws.on('message', whilstSameSocket(function () {
+        this._resetPingTimeout();
```
Comment (SimonSchick) [inline lib/ws/ws.js]: Assuming the `this` context is preserved correctly?

Comment (connor4312) [inline lib/ws/ws.js]: Yea, whilstSameSocket does the `.apply(self`

**Comment by SimonSchick** [comment]
@connor4312 pls publish

**Comment by artdude543** [comment]
When is this being pushed to NPM? 😺

**Comment by SimonSchick** [comment]
@artdude543 now

**Pull request merged**